test/integration/http2_integration_test.cc

```diff
@@ -109,6 +109,26 @@ TEST_P(Http2MetadataIntegrationTest, ProxyMultipleMetadataReachSizeLimit) {
   testEnvoyMultipleMetadataReachSizeLimit();
 }
 
+TEST_P(Http2MetadataIntegrationTest, ProxySmallMetadataInRequest) {
```

Comment:
Also sorry I missed this on prior PRs but you can define the tests inline here as they're only used in http2_integration_test.

I'm trying to fix the thing where we define the code in one place and then run the tests elsewhere :-)

Reply:
Fixed. Sorry it's a little messy because I also moved response tests.